codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/shapevisitor/DafnyToSmithyShapeVisitor.java

```diff
@@ -318,6 +328,18 @@ public String structureShape(final StructureShape shape) {
     final String referenceType = (this.isOptional) ? "&" : "";
 
     final var typeConversionMethodBuilder = new StringBuilder();
+    if (this.isOptional) {
+      final String unAssertedDataSource = dataSource.startsWith("input.(")
+        ? "input"
+        : dataSource;
+      typeConversionMethodBuilder.append(
```

Comment:
You might be able to use regex and just check for `.(%s)` and then split at that. Makes it robust against changing the name `input`.